app/helpers/submissions_helper.rb

```diff
@@ -62,20 +63,29 @@ def get_submissions_table_info(assignment, groupings)
           result = submission.remark_result
         end
         g[:name] = grouping.get_group_name
-        g[:id] = grouping.id
+        if current_user.student?
+          g[:id] = nil
+          g[:name_url] = nil
+          g[:repo_name] = nil
+          g[:repo_url] = nil
+          g[:final_grade] = nil
```

Comment:
If these are the columns which are *not* necessary for students to see, it seems this list should be much bigger.